ystia/yorc
# Feature/gh 554 file store

## Description of the change
Allow to store deployments data into files instead of Consul KV Store
### What I did
- New default implementation for deployments store: file with cache system.
Deployments are stored in the "store" directory under the yorc configurable working directory.

- Update tests to store deployments in a temporary directory.

- Add Upgrade process for Consul Schema to pass to 1.3.0

### How to verify it
You can run a benchmark on a topology definition storage.
```
$ cd deployments
$ go test -run Benchmarch* -bench=. -count 2 -benchtime 15s
```
storage should be around **15** times faster

**On develop;**
BenchmarkDefinitionStore-4             3        5258569786 ns/op
BenchmarkDefinitionStore-4             3        5338326330 ns/op

**On this PR:**
BenchmarkDefinitionStore-4            51         348879640 ns/op
BenchmarkDefinitionStore-4            51         394977759 ns/op

### Description for the changelog

## Applicable Issues
#554
